tests/e2e/websocket/common.hpp

```diff
@@ -0,0 +1,309 @@
+    if (ec) {
+      return Fail(ec, "close");
+    }
+  }
+
+ private:
+  using InternalStream = std::conditional_t<ssl, beast::ssl_stream<beast::tcp_stream>, beast::tcp_stream>;
+  tcp::resolver resolver_;
+  websocket::stream<InternalStream> ws_;
+  beast::flat_buffer buffer_;
+  std::string host_;
+  std::vector<std::string> &received_messages_;
+  std::optional<Credentials> creds_{std::nullopt};
+};
+
+std::unique_ptr<mg::Client> GetBoltClient(const uint16_t bolt_port, const bool use_ssl) {
+  auto client = mg::Client::Connect({.host = "127.0.0.1", .port = bolt_port, .use_ssl = use_ssl});
+  MG_ASSERT(client, "Failed to connect!");
+
+  return client;
+}
+
+inline void CleanDatabase(std::unique_ptr<mg::Client> &client) {
+  MG_ASSERT(client->Execute("MATCH (n) DETACH DELETE n;"));
+  client->DiscardAll();
+}
+
+inline void AddUser(std::unique_ptr<mg::Client> &client) {
+  MG_ASSERT(client->Execute("CREATE USER test IDENTIFIED BY 'testing';"));
+  client->DiscardAll();
+}
+
+inline void AddVertex(std::unique_ptr<mg::Client> &client) {
+  MG_ASSERT(client->Execute("CREATE ();"));
+  client->DiscardAll();
+}
+inline void AddConnectedVertices(std::unique_ptr<mg::Client> &client) {
+  MG_ASSERT(client->Execute("CREATE ()-[:TO]->();"));
+  client->DiscardAll();
+}
+
+inline void RunQueries(std::unique_ptr<mg::Client> &mg_client) {
+  CleanDatabase(mg_client);
+  AddVertex(mg_client);
+  AddVertex(mg_client);
+  AddVertex(mg_client);
+  AddConnectedVertices(mg_client);
+  CleanDatabase(mg_client);
+}
+
+inline void AssertAuthMessage(auto &json_message, const bool success = true) {
+  MG_ASSERT(json_message.at("message").is_string(), "Event is not a string!");
+  MG_ASSERT(json_message.at("success").is_boolean(), "Success is not a boolean!");
+  MG_ASSERT(json_message.at("success").template get<bool>() == success, "Success does not match expected!");
+}
+
+inline void AssertLogMessage(const std::string &log_message) {
+  const auto json_message = nlohmann::json::parse(log_message);
+  if (json_message.contains("success")) {
+    spdlog::info("Received auth message: {}", json_message.dump());
+    AssertAuthMessage(json_message);
+    return;
+  }
+  MG_ASSERT(json_message.at("event").is_string(), "Event is not a string!");
+  MG_ASSERT(json_message.at("event").get<std::string>() == "log", "Event is not equal to `log`!");
+  MG_ASSERT(json_message.at("level").is_string(), "Level is not a string!");
+  MG_ASSERT(std::ranges::count(kSupportedLevels, json_message.at("level")) == 1);
+  MG_ASSERT(json_message.at("message").is_string(), "Message is not a string!");
+}
+
+template <typename TWebsocketClient>
+void TestWebsocketWithoutAnyUsers(std::unique_ptr<mg::Client> &mg_client) {
+  spdlog::info("Starting websocket connection without any users.");
+  auto websocket_client = TWebsocketClient();
+  websocket_client.Connect("127.0.0.1", "7444");
+
+  RunQueries(mg_client);
+  std::this_thread::sleep_for(std::chrono::seconds(1));
```

Comment:
Why  sleep  here?

Reply:
So we give some time to the client to receive messages